### PR TITLE
upkeep/php-reqs: Update PHP compat check to prevent fatal error

### PR DESCRIPTION
**Pull request by Sidsector9**
### Description of the Change

Recently Publisher Media Kit reported a fatal issue caused due to redeclaration of a function across our 10up plugins.
Ref: https://github.com/10up/publisher-media-kit/issues/196

This PR fixes that by using [10up/wp-compat-validation-tool](https://github.com/10up/wp-compat-validation-tool).

### How to test the Change

Try running this PR on a PHP env lesser than the minimum supported version (7.4) and see the plugin gracefully exit with an error message in the admin screen.

### Changelog Entry

> Changed - Update PHP compatibility check by using [10up/wp-compat-validation-tool](https://github.com/10up/wp-compat-validation-tool)

### Credits

Props @Sidsector9 

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Comment by claudiulodro** [comment]
This solves it nicely in my testing. Thanks!

**Review by faisal-alvi — CHANGES_REQUESTED** [review]
@Sidsector9 LGTM but Cypress and Linting workflows are failing, could you please check/address them?

**Comment by jeffpaul** [comment]
@Sidsector9 still seeing Cypress/WP trunk check failing here, can you help look into resolving that?

**Comment by Sidsector9** [comment]
@jeffpaul this is expected because the `composer install` step doesn't exist in the `develop` branch used by build release zip. It is added in this PR. Once this is merged to develop, tests will pass.

All tests are passing locally:

Screenshot 2024-08-12 at 10 46 26 PM

File: .github/workflows/push-asset-readme-update.yml [hunk 1]
```diff
@@ -10,10 +10,10 @@ jobs:
     name: Push to trunk
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout code
```
Comment (jeffpaul) [inline .github/workflows/push-asset-readme-update.yml:13]: any reason this was removed?  they can be helpful when reviewing check runs that fail to see where things are going wrong

Comment (Sidsector9) [reply .github/workflows/push-asset-readme-update.yml:13]: You're correct, removed it mistakenly, fixed it in [bbf74d7](https://github.com/10up/simple-local-avatars/pull/291/commits/bbf74d7544b1d299671f0778499a1b24972bee9f).

**Review by faisal-alvi — APPROVED** [review]

**Pull request merged**